connector/google/google.go

```diff
@@ -0,0 +1,188 @@
+// Package google implements logging in through OpenID Connect providers.
+package google
+import (
+	"golang.org/x/oauth2"
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+	"github.com/coreos/go-oidc"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/oauth2/google"
+	"github.com/coreos/dex/connector"
+	"google.golang.org/api/admin/directory/v1"
+	"io/ioutil"
+)
+// Config holds configuration options for OpenID Connect logins.
+type Config struct {
+	Issuer       string `json:"issuer"`
+	ClientID     string `json:"clientID"`
+	ClientSecret string `json:"clientSecret"`
+	RedirectURI  string `json:"redirectURI"`
+	Scopes []string `json:"scopes"`
+	// Optional list of whitelisted domains when using Google
+	// If this field is nonempty, only users from a listed domain will be allowed to log in
+	HostedDomain string `json:"hostedDomain"`
+	FilePath  string `json:"serviceAccountCredPath"`
+	Subject         string `json:"subject"`
+	GroupPrefix	string `json:"groupPrefix"`
+}
+
+// Open returns a connector which can be used to login users through an upstream
+// OpenID Connect provider.
+func (c *Config) Open(logger logrus.FieldLogger) (conn connector.Connector, err error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	provider, err := oidc.NewProvider(ctx, c.Issuer)
+	if err != nil {
+		cancel()
+		return nil, fmt.Errorf("failed to get provider: %v", err)
+	}
+	scopes := []string{oidc.ScopeOpenID}
+	if len(c.Scopes) > 0 {
+		scopes = append(scopes, c.Scopes...)
+	} else {
+		scopes = append(scopes, "profile", "email")
+	}
+	//creating client for extraxting groups for users
+	clientID := c.ClientID
+	contents, err := ioutil.ReadFile(c.FilePath)
+	if err != nil {
+		fmt.Errorf("Unable to read file: %v", err)
+		return
```

Comment:
this should look like line 39:
`return nil, fmt.Errorf("failed to get provider: %v", err)`